engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1381,6 +1382,26 @@ - (void)setSelectedTextRangeLocal:(UITextRange*)selectedTextRange {
       _selectedTextRange = [selectedTextRange copy];
     }
   }
+
+  if (_originalAutocapitalizationType == UITextAutocapitalizationTypeWords) {
+    NSRange selection = ((FlutterTextRange*)_selectedTextRange).range;
+    BOOL shouldCapitalize = NO;
+    if (selection.location == 0) {
+      shouldCapitalize = YES;
+    } else if (selection.location <= self.text.length) {
```

Comment:
can location be equal to the length? (im asking cuz i don't know. @LongCatIsLooong probably knows)

Reply:
sounds good